### ISSUE TITLE
order of libraries makes Realm bomb on Chrome debug

### ISSUE THREAD

**Issue by mschipperheyn** [realm/realm-js#587]
Ok, spent the morning fighting with Realm to get it to work on React-Native where it seemingly suddenly stopped working. Whenever I activated the Debug Remotely on the IOS Simulator, the app would crash with some error about getting an empty response on /create_session. 

Ultimately, I got it to work by changing the order of libc++.tbd, libz.tbd and libRealmReact.a 

Using rnpm link, libc++.tb ended up above libRealmReact and apparently this created some dependency problem. 

This is how I got it work:

<img width="669" alt="screen shot 2016-09-05 at 13 52 50" src="https://cloud.githubusercontent.com/assets/384959/18254030/1070ba44-7370-11e6-9ad0-cc92782366b8.png">

If it is possible to influence the order in which rnpm link puts libraries, I suggest you adapt accordingly because this is a nightmare to debug. Otherwise, a Troubleshooting doc item might be in order.

**Comment by alazier** [comment]
Sorry you were having difficulties. It seems odd to me that the order things are linked would make any difference. I think perhaps you were hitting a spurious problem, and re-ordering the dependencies caused things to rebuild which fixed the issue. I tried putting libc++ first in a project but this didn't seem to cause me any issues. If you change the order back does the issue re-appear? If not then the issue probably lies elsewhere.

**Comment by mschipperheyn** [comment]
I don't think so. I tried removing build directory, cleaning cache, removing and reinstalling react, upgrading realm js from 0.14.3-rc to 0.14.3 etc. I tried reordering it back and yes, the issue does reappear: Mac OS X, IOS simulator.

<img width="673" alt="screen shot 2016-09-06 at 14 51 35" src="https://cloud.githubusercontent.com/assets/384959/18284730/72b5f3d2-7441-11e6-98a1-3354173f4938.png">
